### ISSUE TITLE
AttributeSetRepository getList() method don't use searchcriteria

### ISSUE THREAD

**Issue by zkliouel** [magento/magento2#8686]
Magento 2.1.4

I need to filter attributset by type but the searchcriteria seem's to be not used...

Code sample : 
```
 $filter[] = $this->filterBuilder
            ->setField('entity_type_id')
            ->setConditionType('eq')
            ->setValue(4)
            ->create();
        
 $searchCriteria = $this->searchCriteriaBuilder
            ->addFilters($filter)
            ->create();

$attributSets = $this->attributeSetRepository->getList($searchCriteria)->getItems();
```
In $attributSets i have all attributesets not filtered...

Taking a look at Magento\Eav\Model\AttributeSetRepository.php
And the getList method don't use searchcriteria to filter collection....

**Comment by tormit** [comment]
+1
Thanks for the issue report. Just stumbled on this myself.

**Comment by vrann** [comment]
@zkliouel The issue is fixed in the latest develop. It was done with the introduction of the major changes to the API filters processing. That's why it's not something that can be backported to the 2.1 with the patch release

**Comment by magento-engcom-team** [comment]
@zkliouel, thank you for your report.
The issue is already fixed in develop branch, 2.2.0
